shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

```diff
@@ -107,18 +106,50 @@
         @Override
         public void createAndroidViewForPlatformView(
             @NonNull PlatformViewsChannel.PlatformViewCreationRequest request) {
-          // API level 19 is required for android.graphics.ImageReader.
+          // API level 19 is required for `android.graphics.ImageReader`.
           ensureValidAndroidVersion(Build.VERSION_CODES.KITKAT);
-          platformViewRequests.put(request.viewId, request);
+
+          if (request == null) {
+            throw new IllegalStateException(
+                "Platform view hasn't been initialized from the platform view channel.");
+          }
+
+          if (!validateDirection(request.direction)) {
```

Comment:
Here too?

Reply:
From the app developer perspective, this is an invalid state. It's also caught by https://github.com/flutter/engine/blob/master/shell/platform/android/io/flutter/embedding/engine/systemchannels/PlatformViewsChannel.java#L107 and propagated as an standard method channel result error.